pkg/tests/ossm/smoke_test.go

```diff
@@ -19,50 +19,195 @@ import (
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
 	"github.com/maistra/maistra-test-tool/pkg/util/oc"
 	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
 	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
 	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
 )
 
-func TestBookinfoInjection(t *testing.T) {
-	NewTest(t).Id("A2").Groups(ARM, Full, Smoke, InterOp, Disconnected).Run(func(t TestHelper) {
+func TestSmoke(t *testing.T) {
+	NewTest(t).Groups(ARM, Full, Smoke, InterOp, Disconnected).Run(func(t TestHelper) {
+		t.Log("Smoke Test for SMCP: deploy, upgrade, bookinfo and uninstall")
 		ns := "bookinfo"
 
 		t.Cleanup(func() {
-			app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		toVersion := env.GetSMCPVersion()
+		fromVersion := toVersion.GetPreviousVersion()
+
+		oc.RecreateNamespace(t, meshNamespace)
+
+		t.NewSubTest(fmt.Sprintf("install bookinfo with smcp %s", fromVersion)).Run(func(t TestHelper) {
+
+			t.LogStepf("Create SMCP %s and verify it becomes ready", fromVersion)
+			assertSMCPDeploysAndIsReady(t, fromVersion)
+
+			t.LogStep("Install bookinfo pods with sidecar and sleep pod without sidecar")
+			app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+
+			t.LogStep("Check whether sidecar is injected in all bookinfo pods")
+			assertSidecarInjectedInAllBookinfoPods(t, ns)
+
+			t.LogStep("Check if bookinfo productpage is running through the Proxy")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns), "sleep",
+					"curl -sI http://productpage:9080",
+					assert.OutputContains(
+						"HTTP/1.1 200 OK",
+						"ProductPage returns 200 OK",
+						"ProductPage didn't return 200 OK"),
+					assert.OutputContains(
+						"server: istio-envoy",
+						"HTTP header 'server: istio-envoy' is present in the response",
+						"HTTP header 'server: istio-envoy' is missing from the response"),
+					assert.OutputContains(
+						"x-envoy-decorator-operation",
+						"HTTP header 'x-envoy-decorator-operation' is present in the response",
+						"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+			})
+
+			t.LogStep("verify proxy startup time. Expected to be less than 10 seconds")
+			t.Log("Jira related: https://issues.redhat.com/browse/OSSM-3586")
+			t.Log("From proxy yaml, verify the time between status.containerStatuses.state.running.startedAt and status.conditions[type=Ready].lastTransitionTime")
+			ratingPod := pod.MatchingSelector("app=ratings", ns)(t, oc.DefaultOC)
+			ratingYaml := oc.GetYaml(t, ns, "pod", ratingPod.Name)
+			t.Log("Validate from YAML, the proxy startup time is less than 10 seconds for ratings pod")
+			validateStartUpProxyTime(t, ratingYaml)
```

Comment:
Should this be a separate subtest? It has its own description, related Jira, and it's a completely separate check than what the rest of this subtest checks.

Reply:
Yes, I can do this and make a separated test

Reply:
Btw, when someone comments on a PR, you should only reply instead of also clicking on "Resolve conversation", so that the person can easily see your reply without having to "Show resolved". The original comment author should be the one to resolve a conversation.